Content.Shared/StepTrigger/StepTriggerSystem.cs

```diff
@@ -29,12 +29,16 @@ public override void Update(float frameTime)
 
     private bool Update(StepTriggerComponent component)
     {
-        if (component.Deleted || !component.Active || component.Colliding.Count == 0)
+        if (component.Deleted ||
+            !component.Active ||
+            component.Colliding.Count == 0 ||
+            !TryComp<TransformComponent>(component.Owner, out var ownerTransform))
             return true;
 
+
         foreach (var otherUid in component.Colliding.ToArray())
         {
-            if (!otherUid.IsValid())
+            if (!otherUid.IsValid() || !TryComp<TransformComponent>(otherUid, out var otherTransform))
```

Comment:
Checking for transform is a bandaid IMO; if you use mousetraps you can replicate the problem.